File: iosApp/iosApp/Services/PasswordServiceAssembler.swift
// Created by web3d4v on 18/11/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation
import web3walletcore

final class PasswordServiceAssembler: AssemblerComponent {

    func register(to registry: AssemblerRegistry) {

        registry.register(scope: .singleton) { resolver -> PasswordService in
            DefaultPasswordService()
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/PasswordServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.password.DefaultPasswordService
import com.sonsofcrypto.web3walletcore.services.password.PasswordService
import smartadapter.internal.extension.name

class PasswordServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(PasswordService::class.name, AssemblerRegistryScope.SINGLETON) {
            DefaultPasswordService()
        }
    }
}